session04/mailroomDictFile.py

```diff
@@ -0,0 +1,126 @@
+        print(name)
+
+def find_donor(name):
+    """ finding donor's name in donor_dict and printing their information """
+    for donor in donor_dict:
+        # case-insenstive compare
+        if name.lower() == donor.lower():
+            return donor
+    return None
+
+def sort_key(item):
+    """ key function used to sort the list by first (not zeroth) item"""
+    return item[1]
+
+
+def print_letter(name, amount):
+    """ writing a thank you letter to donor """
+    print("\nDear {},\n\nThank you for your kind donation of ${}."
+          "\n\nSincerely,\n\nA Team".format(name, amount))
+
+# def letters_for_everyone():
+#     """ writing a thank you letter to all donor """
+#     for name, amounts in donor_list.items():
+#         print("\nDear {},\n\nThank you for your kind donation of ${}."
+#           "\n\nSincerely,\n\nA Team".format(name, amounts))
+
+def letters_for_everyone():    #need to go back and put letters in txt file.
+    """ writing a thank you letter to all donor """
+    for name, amounts in donor_dict.items():
+        new_amounts = []
+        for i in amounts:
+            new_amounts.append('$' + str(i))
+        all_amounts = ", ".join(new_amounts)
+        with open(name + ".txt", 'w') as this_file:
+            this_file.write("\nDear {},\n\nThank you for your kind donation of ${} ({})."
+                  "\n\nSincerely,\n\nA Team".format(name, round(sum(amounts), 2), all_amounts))
+
+
+def print_report():
+    """ printing a report that includes donor names, contribution amounts, sum, and average """
+    report = []
+    for (name, amount) in donor_dict.items():
+        total_amount = sum(amount)
+        number_gifts = len(amount)
+        avg_gift = total_amount / number_gifts
+        report.append((name, total_amount, number_gifts, avg_gift))
+
+    # sorting report data A-Z
+    report.sort(key=sort_key)
+    # printing report in a nicer format
+    print("{:25s} | {:11s} | {:9s} | {:12s}".format(
+        "\nDonor Name", "Total amount", "# of Gifts", "Average amount"))
+    print("-" * 70)
+    for row in report:
+        print("{:25s}   {:11.2f}   {:9d}   {:12.2f}".format(*row))
+
+def send_thank_you():
+    """" Asking user to select from the following options: Send a Thank You, "Create a report" or "quit" """
+    # could use while True:
+    answer = input("\nPlease enter a donor's name."
+                   "\nOr 'list for list of donors' names ")
+    if answer == "list":
+        print_donor_list()
+    else:
+        amounts = float(input("Enter donation amount "))
+        donor = find_donor(answer)
+        if donor is None:
+            donor_dict[answer] = [amounts]   # code on the left is correct (incorrect: donor = (answer, [float(amounts)]))
+        else:
+            donor_dict[donor].append(amounts)   # import pdb; pdb.set_trace()--checking to see why can't use append
+
+        # print a letter to thanks the donor
+        print(print_letter(answer, float(amounts)))
```

Comment:
Have you noticed that you are printing out None to the command line after the letter? This is because print will print whatever is returned by the function you are calling in it. In this case, print_letter is not actually returning anything, so it prints None. Print_letter is doing all of the printing within the function itself, so just a call to print_letter is sufficient to print the letter, no need to call print here.